typespec/Applications.Core/environments.tsp

```diff
@@ -103,24 +110,51 @@ model SecretConfig {
   secret?: string;
 }
 
-@doc("The Cloud providers configuration")
+// ProviderConfigProperties allows to get the additional properties. To ensure that `additionalProperties` is true, we need to extend `Record<unknown>`.
+// Reference: https://github.com/Azure/typespec-azure/blob/main/packages/typespec-autorest/test/additional-properties.test.ts
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("This configuration holds the necessary information to authenticate and interact with a provider for the recipe execution.")
+model ProviderConfigProperties extends Record<unknown> {
+  @doc("Secrets are encrypted, often have fine-grained access control, auditing and are recommended to be used to hold sensitive data.")
+  secrets?: Record<RecipeSecret>;
+}
+
+// EnvironmentVariables allows to get the additional properties. To ensure that `additionalProperties` is true, we need to extend `Record<unknown>`.
+// Reference: https://github.com/Azure/typespec-azure/blob/main/packages/typespec-autorest/test/additional-properties.test.ts
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("The environment variables injected during Terraform Recipe execution for the recipes in the environment.")
+model EnvironmentVariables extends Record<unknown> {
+  @doc("Environment variables containing sensitive information can be fetched from `Applications.Core/SecretStore` resource.")
+  secrets?: Record<RecipeSecret>;
+}
+
+@doc("This secret is used within a recipe. Secrets are encrypted, often have fine-grained access control, auditing and are recommended to be used to hold sensitive data.")
+model RecipeSecret {
```

Comment:
The design doc calls this `ProviderSecret`.

Reply:
This was me here https://github.com/vishwahiremat/radius/pull/1#discussion_r1499673877. @rynowak if you think ProviderSecret makes more sense and we should add new properties for secrets for anything else that we might introduce in the future, then keeping it as ProviderSecret makes sense to me.

I think should have brought this up for a re-review of api spec design if it absolutely needed to change.

Reply:
@lakshmimsft @ytimocin Let's re-review this and `env` in the design review today.